### PR TITLE
Add tests for auth token polling machine

### PR DESCRIPTION
**Pull request by jbranchaud**
This adds some tests to cover the XState machine that polls the server for an auth token. It adds the `SimulatedClock` module recommended by @davidkpiano for advancing timers used by delayed transitions.

polling

### CODE REVIEW

File: setupTests.ts [hunk 1]
```diff
@@ -15,7 +15,7 @@ declare global {
       /**
        *  Match wheter the XState State matches the provided value.
        */
-      toMatchState(state: string): CustomMatcherResult
+      toMatchState(state: string | object): CustomMatcherResult
```
Comment (jbranchaud) [inline setupTests.ts:18]: You can also match against a State object

File: src/machines/__tests__/auth-token-polling-machine.test.ts [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+  )
+
+  pollingService.start()
+
+  await sleep(0)
+
+  expect(pollingService.state).toMatchState({pending: 'scheduleNextPoll'})
+})
+
+test('polls several times for the value', async () => {
```
Comment (jbranchaud) [inline src/machines/__tests__/auth-token-polling-machine.test.ts:81]: This is the most interesting test. It uses the `SimulatedClock` and `sleep` to both advance the clock and to trigger the Invoked Service. I've left some comments to hopefully make that clear.

File: src/utils/test/simulated-clock.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
+// Source: https://github.com/statelyai/xstate/blob/264679d1393095eca54209a790005b3cf48922c7/packages/core/src/SimulatedClock.ts
+// Per XState Discord: https://discord.com/channels/795785288994652170/809564635614150686/898257435750461490
+
+export interface Clock {
+  setTimeout(fn: (...args: any[]) => void, timeout: number): any
+  clearTimeout(id: any): void
+}
+
+export interface SimulatedClock extends Clock {
```
Comment (jbranchaud) [inline src/utils/test/simulated-clock.ts:9]: The `SimulatedClock` pulled from the `xstate` repo.

Recommended here: https://discord.com/channels/795785288994652170/809564635614150686/898257435750461490

**Review by joelhooks — APPROVED** [review]
i'm gonna be giving these xstate-centric tests the side eye and aren't what i'd generally lean towards because it's a LOT of cognitive load. is the juice worth the squeeze? time will tell. 

love seeing the tests, just concerned about the “orthogonal complexity” 😬

**Pull request merged**